assets/js/components/consent-mode/ConsentModeSetupCTAWidget.js

```diff
@@ -0,0 +1,258 @@
+import useViewOnly from '../../hooks/useViewOnly';
+import { WEEK_IN_SECONDS } from '../../util';
+import { CONSENT_MODE_SETUP_CTA_WIDGET_SLUG } from './constants';
+
+const { useSelect, useDispatch } = Data;
+
+function ConsentModeSetupCTAWidget( { Widget, WidgetNull } ) {
+	const [ isSaving, setIsSaving ] = useState( false );
+	const [ saveError, setSaveError ] = useState( null );
+
+	const viewOnlyDashboard = useViewOnly();
+
+	const isConsentModeEnabled = useSelect( ( select ) =>
+		select( CORE_SITE ).isConsentModeEnabled()
+	);
+
+	const settingsURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getAdminURL( 'googlesitekit-settings' )
+	);
+
+	const consentModeDocumentationURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getDocumentationLinkURL( 'consent-mode' )
+	);
+
+	const showTooltip = useShowTooltip( CONSENT_MODE_SETUP_CTA_WIDGET_SLUG );
+	const { isTooltipVisible } = useTooltipState(
+		CONSENT_MODE_SETUP_CTA_WIDGET_SLUG
+	);
+
+	const isDismissed = useSelect( ( select ) =>
+		select( CORE_USER ).isPromptDismissed(
+			CONSENT_MODE_SETUP_CTA_WIDGET_SLUG
+		)
+	);
+	const dismissCount = useSelect( ( select ) =>
+		select( CORE_USER ).getPromptDismissCount(
+			CONSENT_MODE_SETUP_CTA_WIDGET_SLUG
+		)
+	);
+
+	const { setConsentModeEnabled, saveConsentModeSettings } =
+		useDispatch( CORE_SITE );
+	const { dismissPrompt } = useDispatch( CORE_USER );
+	const { navigateTo } = useDispatch( CORE_LOCATION );
+
+	if ( isTooltipVisible ) {
+		return (
+			<Fragment>
+				<WidgetNull />
+				<AdminMenuTooltip
+					title=""
+					content={ __(
+						'You can always enable consent mode from Settings later',
+						'google-site-kit'
+					) }
+					dismissLabel={ __( 'Got it', 'google-site-kit' ) }
+					tooltipStateKey={ CONSENT_MODE_SETUP_CTA_WIDGET_SLUG }
+				/>
+			</Fragment>
+		);
+	}
+
+	const shouldShowWidget =
+		! viewOnlyDashboard &&
+		( isSaving || ( isDismissed === false && ! isConsentModeEnabled ) );
+
+	if ( ! shouldShowWidget ) {
+		return <WidgetNull />;
+	}
+
+	const handleCTAClick = async () => {
+		setSaveError( null );
+		setIsSaving( true );
+
+		setConsentModeEnabled( true );
+		const { error } = await saveConsentModeSettings();
+
+		if ( error ) {
+			setSaveError( error );
+			setConsentModeEnabled( false );
+			setIsSaving( false );
+		} else {
+			await dismissPrompt( CONSENT_MODE_SETUP_CTA_WIDGET_SLUG );
+			navigateTo( `${ settingsURL }#/admin-settings` );
+		}
+	};
+
+	const handleDismissClick = async () => {
+		// For the first two dismissals, we show the notification again in two weeks.
+		if ( dismissCount < 2 ) {
+			showTooltip();
+
+			const twoWeeksInSeconds = WEEK_IN_SECONDS * 2;
+			await dismissPrompt( CONSENT_MODE_SETUP_CTA_WIDGET_SLUG, {
+				expiresInSeconds: twoWeeksInSeconds,
+			} );
+		} else {
+			// For the third dismissal, dismiss permanently.
+			await dismissPrompt( CONSENT_MODE_SETUP_CTA_WIDGET_SLUG );
+		}
+	};
+
+	return (
+		<div className="googlesitekit-widget-context">
```

Comment:
I've directly reused the widget context and area classes here - in a future update we should look at how we can restructure this to take a common approach to this and the Audience Segmentation banner (and any others which follow the same design pattern).